core/arch/arm/include/kernel/generic_boot.h

```diff
@@ -31,10 +31,12 @@
 
 #if defined(CFG_WITH_ARM_TRUSTED_FW)
 uint32_t cpu_on_handler(uint32_t a0, uint32_t a1);
-uint32_t *generic_boot_init_primary(uint32_t pageable_part);
+uint32_t *generic_boot_init_primary(uint32_t pageable_part, uint32_t unused,
+				    uint32_t fdt);
```

Comment:
With `paddr_t` we can't guarantee that it will fit in a single register so it would sometime use a different calling convention if used in the arguments of a function. In C that's not a problem, but if the functions are called from assembly it's painful. For our purposes we can assume that `unsinged long` always fits in a single register.